examples/all-clusters-app/all-clusters-common/src/operational-state-delegate-impl.cpp

```diff
@@ -17,106 +17,150 @@
+    else
+    {
+        err.Set(to_underlying(ErrorStateEnum::kUnableToCompleteOperation));
+    }
 }
 
-void OperationalStateDelegate::GetCurrentOperationalError(GenericOperationalError & error)
+void GenericOperationalStateDelegateImpl::HandleStopStateCallback(GenericOperationalError & err)
 {
-    error = mOperationalError;
+    // placeholder implementation
+    auto error = mServer->SetOperationalState(to_underlying(OperationalStateEnum::kStopped));
+    if (error == CHIP_NO_ERROR)
+    {
+        err.Set(to_underlying(ErrorStateEnum::kNoError));
+    }
+    else
+    {
+        err.Set(to_underlying(ErrorStateEnum::kUnableToCompleteOperation));
+    }
 }
 
-void OperationalStateDelegate::GetCurrentPhase(app::DataModel::Nullable<uint8_t> & phase)
-{
-    phase = mOperationalPhase;
-}
+// Init Operational State cluster
 
-void OperationalStateDelegate::GetCountdownTime(app::DataModel::Nullable<uint32_t> & time)
-{
-    time = mCountdownTime;
-}
+static OperationalState::Instance * gOperationalStateInstance = nullptr;
+static OperationalStateDelegate * gOperationalStateDelegate   = nullptr;
 
-CHIP_ERROR OperationalStateDelegate::GetOperationalStateAtIndex(size_t index, GenericOperationalState & operationalState)
+void OperationalState::Shutdown()
 {
-    if (index > mOperationalStateList.size() - 1)
+    if (gOperationalStateInstance != nullptr)
     {
-        return CHIP_ERROR_NOT_FOUND;
+        delete gOperationalStateInstance;
+        gOperationalStateInstance = nullptr;
     }
-    operationalState = mOperationalStateList[index];
-    return CHIP_NO_ERROR;
-}
-
-CHIP_ERROR OperationalStateDelegate::GetOperationalPhaseAtIndex(size_t index, GenericOperationalPhase & operationalPhase)
-{
-    if (index > mOperationalPhaseList.size() - 1)
+    if (gOperationalStateDelegate != nullptr)
     {
-        return CHIP_ERROR_NOT_FOUND;
+        delete gOperationalStateDelegate;
+        gOperationalStateDelegate = nullptr;
     }
-    operationalPhase = mOperationalPhaseList[index];
-    return CHIP_NO_ERROR;
 }
 
-void OperationalStateDelegate::HandlePauseStateCallback(GenericOperationalError & err)
+void emberAfOperationalStateClusterInitCallback(chip::EndpointId endpointId)
 {
-    // placeholder implementation
-    mOperationalState = to_underlying(OperationalStateEnum::kPaused);
-    err.Set(to_underlying(ErrorStateEnum::kNoError));
-}
+    VerifyOrDie(endpointId == 1); // this cluster is only enabled for endpoint 1.
+    VerifyOrDie(gOperationalStateInstance == nullptr && gOperationalStateDelegate == nullptr);
 
-void OperationalStateDelegate::HandleResumeStateCallback(GenericOperationalError & err)
-{
-    // placeholder implementation
-    mOperationalState = to_underlying(OperationalStateEnum::kRunning);
-    err.Set(to_underlying(ErrorStateEnum::kNoError));
+    gOperationalStateDelegate = new OperationalStateDelegate;
+    gOperationalStateInstance = new Instance(gOperationalStateDelegate, 0x01, Clusters::OperationalState::Id);
```

Comment:
what does '0x01' mean? it would be useful to use a variable with a meaningful name, for code readability

Reply:
I have explicitly created a var to make it more readable.